lib/deref/hasValidParentReference.js

```diff
@@ -0,0 +1,33 @@
+var __parent = require("./../internal/parent");
+var __invalidated = require("./../internal/invalidated");
+module.exports = function fromWhenceYeCame() {
+    var reference = this._referenceContainer;
```

Comment:
Worth clarifying what "this" is assumed to be in this context, since it's just a standalone function.

On the other hand, if this is not on a critical path (that is, it's not called 1000s of times for a typical use case), could we just have this accept a model, and pass it through from the Model.prototype._hasValidParentReference() instead, like the walk implementations.

It just seemed a little odd to see a standalone function, reference "this", when "this" can only really ever be a Model instance, or something which shares the Model interface.

I don't think we need to change this for this PR (since it looks like we do it in quite a few places for stuff under lib), but worth considering for future development/code organization.